src/frontends/pytorch/src/transforms/rfftn_complex_replacer.cpp

```diff
@@ -0,0 +1,135 @@
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+using namespace ov::pass;
+using namespace ov::op;
+
+RFFTNComplexReplacer::RFFTNComplexReplacer() {
+    auto fft_op = pattern::wrap_type<ov::op::util::FrameworkNode>();
+    ov::matcher_pass_callback rfftn_callback = [](pattern::Matcher& m) {
+        // Pattern: aten::fft_rfftn -> {aten::real, aten::imag}
+        // "aten::fft_rfftn(Tensor self, int[1]? s=None, int[1]? dim=None, str? norm=None) -> Tensor"
+        auto rfftn_op = cast_fw_node(m.get_match_root(), "aten::fft_rfftn");
+        if (!rfftn_op) {
+            return false;
+        }
+        auto const_neg_1 = v0::Constant::create(element::i32, Shape{}, {-1});
+        auto const_0 = v0::Constant::create(element::i32, Shape{}, {0});
+        auto const_1 = v0::Constant::create(element::i32, Shape{}, {1});
+
+        auto input = rfftn_op->input_value(0);
+        auto input_shape = std::make_shared<v3::ShapeOf>(input, element::i32);
+        auto input_rank = std::make_shared<v3::ShapeOf>(input_shape, element::i32);
+        auto input_rank_scalar = std::make_shared<v0::Squeeze>(input_rank);
+
+        auto node_s_input = concat_list_construct(rfftn_op->input_value(1)).get_node_shared_ptr();
+
+        bool dim_use_default = is_none_node(rfftn_op->input_value(2).get_node_shared_ptr());
+        bool s_use_default = is_none_node(rfftn_op->input_value(1).get_node_shared_ptr());
+
+        std::shared_ptr<ov::Node> dim;
+        if (!dim_use_default) {
+            dim = std::make_shared<v0::Convert>(concat_list_construct(rfftn_op->input_value(2)), element::i32);
+        } else if (!s_use_default) {
+            auto s_len = std::make_shared<v3::ShapeOf>(node_s_input, element::i32);
+            auto slice_start = std::make_shared<v1::Subtract>(input_rank, s_len);
+            auto slice_start_scalar = std::make_shared<v0::Squeeze>(slice_start);
+            dim = std::make_shared<v0::Range>(slice_start_scalar, input_rank_scalar, const_1);
+        } else {
+            dim = std::make_shared<v0::Range>(const_0, input_rank_scalar, const_1);
+        }
+
+        std::shared_ptr<ov::Node> s;
+        if (!s_use_default) {
+            auto full_s_cond = std::make_shared<v1::Equal>(node_s_input, const_neg_1);
+            auto full_s_values = std::make_shared<v8::Gather>(input_shape, dim, const_0);
+            s = std::make_shared<v1::Select>(full_s_cond, full_s_values, node_s_input);
+        } else {
+            s = std::make_shared<v8::Gather>(input_shape, dim, const_0);
+        }
+
+        std::string norm;
+        if (const auto& fw_node_mode = std::dynamic_pointer_cast<ov::op::util::FrameworkNode>(
+                rfftn_op->input_value(3).get_node_shared_ptr())) {
+            const auto& attrs = fw_node_mode->get_attrs();
+            if (attrs.find("string_value") != attrs.end()) {
+                norm = attrs.at("string_value");
+            } else {
+                norm = "backward";
```

Comment:
Is this the case when input is none node?

Reply:
Yes